src/Dom/Document.php

```diff
@@ -1021,6 +1023,33 @@ private function restore_doctype_node( $html ) {
 		return preg_replace( self::HTML_RESTORE_DOCTYPE_PATTERN, '\1!\3\4>', $html, 1 );
 	}
 
+	/**
+	 * Normalizes HTML attributes to be HTML5 compatible.
+	 *
+	 * Conditionally removes html[xmlns], and converts html[xml:lang] to html[lang].
+	 */
+	private function normalize_html_attributes() {
+		$html = $this->getElementsByTagName( 'html' )->item( 0 );
+		if ( ! $html->attributes ) {
+			return;
+		}
+
+		$xmlns                = $html->attributes->getNamedItem( 'xmlns' );
+		$xmlns_value_to_strip = 'http://www.w3.org/1999/xhtml';
+		if ( $xmlns && $xmlns_value_to_strip === $xmlns->nodeValue ) {
+			$html->removeAttributeNode( $xmlns );
+		}
+
+		$xml_lang = $html->attributes->getNamedItem( 'xml:lang' );
+		if ( $xml_lang ) {
+			if ( ! $html->attributes->getNamedItem( 'lang' ) && $xml_lang->nodeValue ) {
+				// Move the html[xml:lang] value to html[lang].
+				$html->setAttribute( 'lang', $xml_lang->nodeValue );
```

Comment:
@todo: handle case of `lang=""`.

If `<head xml:lang="es" lang="">`, this will produce `<head lang="">`. 

Currently, it won't copy the `xml:lang` value to `lang` if `lang=""`.

Reply:
Applied in https://github.com/ampproject/amp-wp/pull/4194/commits/93d96f4a9e92e021597ecae524bf9d476f84f552